x/configuration/client/cli/tx.go

```diff
@@ -40,6 +44,92 @@ var defaultDuration, _ = time.ParseDuration("1h")
 const defaultRegex = "^(.*?)?"
 const defaultNumber = 1
 
+func getCmdUpdateFee(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "update-fee",
+		Short:   "update fee parameter(s) using key=value pair(s) noting that all numeric values should be specified in euros",
+		Example: "iovnscli tx configuration update-fee --from iovSAS --parameter fee_coin_price=0.29 --parameter register_domain_default=12",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc).WithBroadcastMode(flags.BroadcastBlock)
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			txBuilder := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
+			// query the existing fee configuration
+			path := fmt.Sprintf("custom/%s/%s", types.StoreKey, types.QueryFees)
+			resp, _, err := cliCtx.Query(path)
+			if err != nil {
+				return err
+			}
+			var jsonResp types.QueryFeesResponse
+			err = queries.DefaultQueryDecode(resp, &jsonResp)
+			if err != nil {
+				return err
+			}
+			fees := jsonResp.Fees
+			// set new values via json tag https://gist.github.com/lelandbatey/a5c957b537bed39d1d6fb202c3b8de06
+			v := reflect.ValueOf(fees).Elem()
+			findJsonName := func(t reflect.StructTag) string {
+				if jt, ok := t.Lookup("json"); ok {
+					return strings.Split(jt, ",")[0]
+				}
+				panic(fmt.Errorf("tag provided does not define a json tag"))
+			}
+			fieldNames := map[string]int{}
+			for i := 0; i < v.NumField(); i++ {
+				typeField := v.Type().Field(i)
+				tag := typeField.Tag
+				jname := findJsonName(tag)
+				fieldNames[jname] = i
+			}
+			// read the fee parameter pair(s)
+			pairs, err := cmd.Flags().GetStringArray("parameter")
+			if err != nil {
+				return err
+			}
+			// iterate over the pair(s) and set the fee parameter value
+			for _, raw := range pairs {
+				split := strings.Split(raw, "=")
+				if len(split) < 1 {
+					return fmt.Errorf("invalid formatted value: %s", raw)
+				}
+				key := split[0]
+				value := strings.Join(split[1:], "=")
```

Comment:
@fdymylja , why `value := strings.Join(split[1:], "=")` and not `value := split[1]`?  (I lifted the code from you signutil command.)

Reply:
arrays in cobra are passed to CLI flags as (considering pair has 'p' as shorthand)
`-p hello=hi -p name=frojdi
`
they are then retrieved in code as `pairs := flag.GetStringArray` which resolves pair as  `string[]{"hello=hi", "name=frojdi"}`

hence we split them to get an array of key values

and then we split each element of the array to get the respective key values

Reply:
it's done in case the value contains `=`, example:

```
Pair {
Key: "function",
Value: "y=x",
}
```

If we did not join them the `split[1]` value would be only `y `and not `y=x`